bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusRelayWithInputHandler.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.velbus.internal.handler;
+
+import static org.openhab.binding.velbus.internal.VelbusBindingConstants.*;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.velbus.internal.VelbusChannelIdentifier;
+import org.openhab.binding.velbus.internal.packets.VelbusButtonPacket;
+import org.openhab.binding.velbus.internal.packets.VelbusPacket;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.CommonTriggerEvents;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link VelbusRelayWithInputHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniel Rosengarten - Initial contribution
+ */
+@NonNullByDefault
+public class VelbusRelayWithInputHandler extends VelbusRelayHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(Arrays.asList(THING_TYPE_VMB1RYS));
+
+    private static final StringType PRESSED = new StringType("PRESSED");
+    private static final StringType LONG_PRESSED = new StringType("LONG_PRESSED");
+
+    public VelbusRelayWithInputHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+
+        VelbusBridgeHandler velbusBridgeHandler = getVelbusBridgeHandler();
+        if (velbusBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
+        }
+
+        if (isButtonChannel(channelUID) && command instanceof StringType) {
+            StringType stringTypeCommand = (StringType) command;
+
+            if (stringTypeCommand.equals(PRESSED) || stringTypeCommand.equals(LONG_PRESSED)) {
+                VelbusButtonPacket packet = new VelbusButtonPacket(getModuleAddress().getChannelIdentifier(channelUID));
+
+                try {
+                    packet.Pressed();
+                    velbusBridgeHandler.sendPacket(packet.getBytes());
+                    Thread.sleep(20);
+
+                    if (stringTypeCommand.equals(LONG_PRESSED)) {
+                        packet.LongPressed();
+                        velbusBridgeHandler.sendPacket(packet.getBytes());
+                        Thread.sleep(20);
```

Comment:
I don't think you are allowed to let the thread sleep here.
But you could use the scheduler, like [here](https://github.com/openhab/openhab-addons/blob/5ec535f37bef6f2491da64cd884af02b454e7d03/bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusBridgeHandler.java#L155)

Reply:
I d'idn't check this part of your code, the 20 ms delay is useless then. I've remove this from my code.